src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1582,6 +1572,16 @@ public override BoundNode VisitForStatement(BoundForStatement node)
             return result;
         }
 
+        public override BoundNode VisitForEachStatement(BoundForEachStatement node)
+        {
+            // NOTE: iteration variables are not declared or assigned
+            //       before the collection expression is evaluated 
+            var result = base.VisitForEachStatement(node);
+
+            // NOTE: do not report unused iteration variables. They are always considered used.
```

Comment:
Please move this comment to the line `NoteWrite` in `VisitForEachIterationVariables`, which is what I think implements this policy.